src/main/java/com/hyunec/cosmicbaseball/entity/BaseballGame.java

```diff
@@ -1,71 +1,97 @@
 package com.hyunec.cosmicbaseball.entity;
 
 import com.hyunec.cosmicbaseball.handler.ResultHandler;
-import org.springframework.stereotype.Component;
-import java.util.Arrays;
+import com.hyunec.cosmicbaseball.util.BattingResult;
 import java.util.List;
 import java.util.Random;
+import org.springframework.stereotype.Component;
 
 @Component
 public class BaseballGame {
-    private int strikeCount;
-    private int ballCount;
-    private final Random random;
 
-    private final ResultHandler resultHandler;
+  private int strikeCount;
+  private int ballCount;
+  private String lastResult;
+  private final Random random;
+  private List<String> possibleResults;
 
-    private BaseballGame(ResultHandler resultHandler) {
-        this.strikeCount = 0;
-        this.ballCount = 0;
-        this.random = new Random();
-        this.resultHandler = resultHandler;
-    }
+  private final ResultHandler resultHandler;
 
-    //스윙했을 때 돌아올 수 있는 값들 담아놓음.
-    public String swing() {
-        List<String> possibleResults = Arrays.asList("스트라이크", "볼", "안타");
-        int index = random.nextInt(possibleResults.size());
-        return possibleResults.get(index);
-    }
+  public BaseballGame(ResultHandler resultHandler, List<String> possibleResults
+  ) {
+    this.strikeCount = 0;
+    this.ballCount = 0;
+    this.lastResult = "";
+    this.random = new Random();
+    this.resultHandler = resultHandler;
+    this.possibleResults = possibleResults;
+  }
 
-    public void processResult(String result) {
+  public BattingResult swing() {
+    //스윙하는 행위만 담기
+    return BattingResult.getRandomResult();
+  }
```

Comment:
```suggestion
    /**
     * 스윙하는 행위만 담기
     */
    public BattingResult swing() {
        return BattingResult.getRandomResult();
    }
```

기본적으로 javadoc 스타일의 주석을 권장 합니다.